### PR TITLE
Top ups in dapp

### PR DESCRIPTION
**Pull request by r-czajkowski**
Ref: https://github.com/keep-network/keep-core/issues/1898

This PR adds support for top-ups in a keep token dashboard.

<details>
  <summary>Screenshoots</summary>

obraz
obraz
obraz

</details>

### CODE REVIEW

**Comment by dimpar** [comment]
Switching between Metamask accounts doesn't work properly. Upon selection I can see the below error and it will switch only after page reloads.

Screenshot 2020-07-30 at 11 33 26

**Comment by dimpar** [comment]
I think one of the things that can be improved a bit is disabling `commit top-up` or `add keep` button until the initialization period is over. As for now they are still clickable but MetaMask will reject that transaction.

**Comment by dimpar** [comment]
I don't think we need "Clickable finger palm" over Current / New 

Screenshot 2020-07-30 at 11 57 14

**Comment by r-czajkowski** [comment]
> I think one of the things that can be improved a bit is disabling `commit top-up` or `add keep` button until the initialization period is over. As for now they are still clickable but MetaMask will reject that transaction.

Yeah, nice catch!

**Comment by r-czajkowski** [comment]
> I don't think we need "Clickable finger palm" over Current / New
> 
> Screenshot 2020-07-30 at 11 57 14

Implemented based on [KEEP Design system](https://www.figma.com/file/ARAMFLMOPal0JewjvlZGhX/Keep-Design-System?node-id=1168%3A80) 🤔

**Comment by dimpar** [comment]
Yeah, to be clear, I'm not talking about changing colors but rather about "navigation hand / palm" whatever you call it :) Usually you expect to be navigated to somewhere upon click, but in this case it just change color.

https://upload.wikimedia.org/wikipedia/commons/thumb/3/39/Pointing_hand_cursor_vector.svg/1200px-Pointing_hand_cursor_vector.svg.png

**Comment by r-czajkowski** [comment]
> Yeah, to be clear, I'm not talking about changing colors but rather about "navigation hand / palm" whatever you call it :) Usually you expect to be navigated to somewhere upon click, but in this case it just change color.
> 
> https://upload.wikimedia.org/wikipedia/commons/thumb/3/39/Pointing_hand_cursor_vector.svg/1200px-Pointing_hand_cursor_vector.svg.png

yeah, I see 👍

**Comment by dimpar** [comment]
Scenario:
Delegated 100k from Grant 1 to 0xAAA. But right after I canceled that. And then again I've delegated 100k from Grant 1 to 0xAAA and it crashed:

Screenshot 2020-07-30 at 12 19 11

**Comment by dimpar** [comment]
Problem: cannot undelegate tokens

Scenario:
Delegate tokens from Grant to 0xAAA -> Add Keep and Commit-topup. Then Undelegate tokens, but MetaMask won't let you.

Screenshot 2020-07-30 at 13 32 35

**Comment by dimpar** [comment]
Problem: Can't recover undelegated tokens

Scenario: Delegate from Grant -> Undelegate. Then tried Recover but it failed.

Screenshot 2020-07-30 at 13 50 54 (3)

File: solidity/dashboard/src/hooks/useFetchData.js [hunk 1]
```diff
@@ -40,6 +43,7 @@ export const useFetchData = (
           dispatch({ type: FETCH_REQUEST_SUCCESS, payload: data })
       })
       .catch((error) => {
+        console.log("error", error)
```
Comment (dimpar) [inline solidity/dashboard/src/hooks/useFetchData.js:46]: console.log

File: solidity/dashboard/src/services/top-ups.service.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+  return availableTopUps
+}
+
+const reduceByOperator = (result, event) => {
+  const {
+    returnValues: { operator },
+  } = event
+
+  ;(result[operator] = result[operator] || []).push(event)
```
Comment (dimpar) [inline solidity/dashboard/src/services/top-ups.service.js:67]: semicolon in a weird plac

File: solidity/dashboard/src/services/top-ups.service.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+const reduceAmount = (result, { returnValues: { topUp } }) => {
+  return add(result, topUp)
+}
+
+const filterByAfterLatestCompletedTopUp = (latestTopUpCompletedEvent) => (
+  initiatedEvent
+) => {
+  const isAfterLatestCompleteedTopUpBlock =
```
Comment (dimpar) [inline solidity/dashboard/src/services/top-ups.service.js:79]: typo: isAfterLatestCompleteedTopUpBlock -> isAfterLatestCompletedTopUpBlock

File: solidity/dashboard/src/services/top-ups.service.js (hunk 2 above)
Comment (r-czajkowski) [reply solidity/dashboard/src/services/top-ups.service.js:67]: eslint...

**Comment by pdyraga** [comment]
The problems with canceled stake and top-ups are probably related to https://github.com/keep-network/keep-core/pull/1949.

**Comment by dimpar** [comment]
Regarding https://github.com/keep-network/keep-core/pull/1945#issuecomment-666260428

I can switch accounts now just fine, but I still can see the same error in the console.

Also, what I noticed is that the flow changed a bit. Not sure if that was intentional, but upon switching MM accounts it won't navigate me to the wallets selection page (MM, ledger, trezor). Only after the page reload I am able to select a wallet or clicking at the tokenDashboard logo.

**Comment by dimpar** [comment]
Problem: Can't commit top-up after adding tokens from Grants

Scenario 1: After init period is over -> add Keep -> you will see a row in top-ups table below. Click enabled "commit top-up" button. MM will reject you. 

Scenario 1: After init period is over -> add Keep -> you will see a row in top-ups table below with enabled button. Just refresh the page without committing. Button is disabled now.

**Comment by dimpar** [comment]
Regarding: https://github.com/keep-network/keep-core/pull/1945#issuecomment-666274506

In happy path flow:

Delegate tokens (status is pending) -> add keep (available Top-up appear) -> commit top-up. It will reject you. Then wait until the init period is over. Commit top-up button is still disabled.

**Comment by dimpar** [comment]
Problem: During init period it seems that I can skip 'Submit top-up' phase and add keep directly. MM will allow me to do so

**Comment by dimpar** [comment]
Problem: Top-ups "disappear" upon delegation.

Scenario: In this example I've delegated 700k tokens -> added 500k tokens but did NOT commit -> undelegated. In the view below it is sort of uncertain where'd 500k top up went. Only if you click "Recover" then 500k will be added to the total available and everything sums up to 10M

Screenshot 2020-07-31 at 15 05 40

**Pull request merged**